pkg/controller/flow/reconcile.go

```diff
@@ -0,0 +1,360 @@
+
+// What field do we assume Object Reference exports as a resolvable target
+const targetFieldName = "domainInternal"
+
+const (
+	// SuccessSynced is used as part of the Event 'reason' when a Flow is synced
+	SuccessSynced = "Synced"
+
+	// MessageResourceSynced is the message used for an Event fired when a Flow
+	// is synced successfully
+	MessageResourceSynced = "Flow synced successfully"
+)
+
+var (
+	flowControllerKind = v1alpha1.SchemeGroupVersion.WithKind("Flow")
+)
+
+// Reconcile compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the Flow resource
+// with the current status of the resource.
+func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	glog.Infof("Reconciling flow %v", request)
+	flow := &v1alpha1.Flow{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, flow)
+
+	if errors.IsNotFound(err) {
+		glog.Errorf("could not find flow %v\n", request)
+		return reconcile.Result{}, nil
+	}
+
+	if err != nil {
+		glog.Errorf("could not fetch Flow %v for %+v\n", err, request)
+		return reconcile.Result{}, err
+	}
+
+	original := flow.DeepCopy()
+
+	flow.Status.InitializeConditions()
+
+	// Reconcile this copy of the Flow and then write back any status
+	// updates regardless of whether the reconcile error out.
+	err = r.reconcile(flow)
+	if equality.Semantic.DeepEqual(original.Status, flow.Status) {
+		// If we didn't change anything then don't call updateStatus.
+		// This is important because the copy we loaded from the informer's
+		// cache may be stale and we don't want to overwrite a prior update
+		// to status with this stale state.
+	} else if _, err := r.updateStatus(flow); err != nil {
+		glog.Warningf("Failed to update flow status: %v", err)
+		return reconcile.Result{}, err
+	}
+
+	// Requeue if the resource is not ready:
+	return reconcile.Result{Requeue: !flow.Status.IsReady()}, err
+}
+
+func (r *reconciler) reconcile(flow *v1alpha1.Flow) error {
+	// See if the flow has been deleted
+	accessor, err := meta.Accessor(flow)
+	if err != nil {
+		log.Fatalf("Failed to get metadata: %s", err)
+	}
+	deletionTimestamp := accessor.GetDeletionTimestamp()
+	glog.Infof("DeletionTimestamp: %v", deletionTimestamp)
+
+	target, err := r.resolveActionTarget(flow.Namespace, flow.Spec.Action)
+	if err != nil {
+		glog.Warningf("Failed to resolve target %v : %v", flow.Spec.Action, err)
+		flow.Status.PropagateActionTargetResolved(corev1.ConditionFalse, "ActionTargetNotResolved", err.Error())
+		return err
+	}
+
+	flow.Status.PropagateActionTargetResolved(corev1.ConditionTrue, "ActionTargetResolved", fmt.Sprintf("Resolved to: %q", target))
+
+	// Ok, so target is the underlying k8s service (or URI if so specified) that we want to target
+	glog.Infof("Resolved Target to: %q", target)
+
+	// Reconcile the Channel. Creates a channel that is the target that the Feed will use.
+	// TODO: We should reuse channels possibly.
```

Comment:
Looks like the current code already reuses a channel with the same name. Is this `TODO` still needed?

Reply:
sorry, it could re-use a channel (not by name but type or something like that) and only create a subscription. Right now it creates 1:1 (channel:subscription). Tried to clarify the comment.